diff --git a/node_modules/expo-modules-core/common/cpp/fabric/ExpoViewProps.cpp b/node_modules/expo-modules-core/common/cpp/fabric/ExpoViewProps.cpp
index ae99fd9..1b4ba1c 100644
--- a/node_modules/expo-modules-core/common/cpp/fabric/ExpoViewProps.cpp
+++ b/node_modules/expo-modules-core/common/cpp/fabric/ExpoViewProps.cpp
@@ -17,10 +17,18 @@ std::unordered_map<std::string, folly::dynamic> propsMapFromProps(const ExpoView
 
   // Iterate over values in the raw props object.
   // Note that it contains only updated props.
+  /*
   rawProps.iterateOverValues([&propsMap](react::RawPropsPropNameHash hash, const char *name, const react::RawValue &value) {
     std::string propName(name);
     propsMap[propName] = (folly::dynamic)value;
   });
+    */
+  const auto& dynamic = static_cast<folly::dynamic>(rawProps);
+
+  for (const auto& pair : dynamic.items()) {
+    std::string propName = pair.first.getString();
+    propsMap[propName] = (folly::dynamic)pair.second;
+  }
 
   return propsMap;
 }
