diff --git a/node_modules/react-native/Libraries/Components/Pressable/Pressable.d.ts b/node_modules/react-native/Libraries/Components/Pressable/Pressable.d.ts
index 63a378d..188e389 100644
--- a/node_modules/react-native/Libraries/Components/Pressable/Pressable.d.ts
+++ b/node_modules/react-native/Libraries/Components/Pressable/Pressable.d.ts
@@ -20,6 +20,7 @@ import {
 import {View} from '../View/View';
 import {AccessibilityProps} from '../View/ViewAccessibility';
 import {ViewProps} from '../View/ViewPropTypes';
+import {TVParallaxProperties} from '../../../types/public/ReactNativeTVTypes';
 
 export interface PressableStateCallbackType {
   readonly pressed: boolean;
@@ -159,6 +160,13 @@ export interface PressableProps
    * Duration (in milliseconds) to wait after press down before calling onPressIn.
    */
   unstable_pressDelay?: number | undefined;
+
+  /**
+   * *(Apple TV only)* Object with properties to control Apple TV parallax effects.
+   *
+   * @platform ios
+   */
+  tvParallaxProperties?: TVParallaxProperties;
 }
 
 // TODO use React.AbstractComponent when available
diff --git a/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.h b/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.h
index 7107a8c..afea86c 100644
--- a/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.h
+++ b/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.h
@@ -25,7 +25,7 @@ NS_ASSUME_NONNULL_BEGIN
 /**
  * UIView class for <View> component.
  */
-@interface RCTViewComponentView : UIView <RCTComponentViewProtocol, RCTTouchableComponentViewProtocol> {
+@interface RCTViewComponentView : UIView <RCTComponentViewProtocol, RCTTouchableComponentViewProtocol, UIGestureRecognizerDelegate> {
  @protected
   facebook::react::LayoutMetrics _layoutMetrics;
   facebook::react::SharedViewProps _props;
diff --git a/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm b/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm
index 6a62196..749cf83 100644
--- a/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm
+++ b/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm
@@ -64,6 +64,7 @@ const CGFloat BACKGROUND_COLOR_ZPOSITION = -1024.0f;
   NSMutableArray<UIView *> *_reactSubviews;
   BOOL _motionEffectsAdded;
   UILongPressGestureRecognizer * _pressRecognizer;
+  UILongPressGestureRecognizer * _longPressRecognizer;
   NSSet<NSString *> *_Nullable _propKeysManagedByAnimated_DO_NOT_USE_THIS_IS_BROKEN;
   UIView *_containerView;
   BOOL _useCustomContainerView;
@@ -158,6 +159,11 @@ const CGFloat BACKGROUND_COLOR_ZPOSITION = -1024.0f;
   }
 }
 
+// Press recognizer should allow long press recognizer to work (but not the reverse)
+- (BOOL)gestureRecognizer:(UIGestureRecognizer *)gestureRecognizer shouldRecognizeSimultaneouslyWithGestureRecognizer:(UIGestureRecognizer *)otherGestureRecognizer {
+  return gestureRecognizer == _pressRecognizer;
+}
+
 #if TARGET_OS_TV
 
 #pragma mark - Apple TV methods
@@ -267,59 +273,106 @@ const CGFloat BACKGROUND_COLOR_ZPOSITION = -1024.0f;
     [[NSNotificationCenter defaultCenter] postNavigationBlurEventWithTag:@(self.tag) target:@(self.tag)];
 }
 
-- (void)sendSelectNotification:(UIGestureRecognizer *)recognizer
+- (void)sendLongSelectStartNotification
 {
-    [[NSNotificationCenter defaultCenter] postNavigationPressEventWithType:RCTTVRemoteEventSelect keyAction:RCTTVRemoteEventKeyActionUp tag:@(self.tag) target:@(self.tag)];
+    [[NSNotificationCenter defaultCenter] postNavigationPressEventWithType:RCTTVRemoteEventLongSelect keyAction:RCTTVRemoteEventKeyActionDown tag:@(self.tag) target:@(self.tag)];
 }
 
-- (void)sendLongSelectNotification:(UIGestureRecognizer *)recognizer
+- (void)sendLongSelectEndNotification
 {
-    [[NSNotificationCenter defaultCenter] postNavigationPressEventWithType:RCTTVRemoteEventLongSelect keyAction:recognizer.eventKeyAction tag:@(self.tag) target:@(self.tag)];
+    [[NSNotificationCenter defaultCenter] postNavigationPressEventWithType:RCTTVRemoteEventLongSelect keyAction:RCTTVRemoteEventKeyActionUp tag:@(self.tag) target:@(self.tag)];
 }
 
-- (void)animatePress
+- (void)animatePressIn
 {
   if (_tvParallaxProperties.enabled == YES) {
-    float magnification = _tvParallaxProperties.magnification;
     float pressMagnification = _tvParallaxProperties.pressMagnification;
 
     // Duration of press animation
     float pressDuration = _tvParallaxProperties.pressDuration;
 
-    // Delay of press animation
-    float pressDelay = _tvParallaxProperties.pressDelay;
+    [UIView animateWithDuration:(pressDuration/2)
+                     animations:^{
+      self.transform = CGAffineTransformMakeScale(pressMagnification, pressMagnification);
+    }
+                     completion:^(__unused BOOL finished1){}];
+  }
+}
 
-    [[NSRunLoop currentRunLoop] runUntilDate:[NSDate dateWithTimeIntervalSinceNow:pressDelay]];
+- (void) animatePressOut
+{
+  if (_tvParallaxProperties.enabled == YES) {
+    float magnification = _tvParallaxProperties.magnification;
+
+    // Duration of press animation
+    float pressDuration = _tvParallaxProperties.pressDuration;
 
     [UIView animateWithDuration:(pressDuration/2)
                      animations:^{
-      self.transform = CGAffineTransformMakeScale(pressMagnification, pressMagnification);
+      self.transform = CGAffineTransformMakeScale(magnification, magnification);
     }
-                     completion:^(__unused BOOL finished1){
-      [UIView animateWithDuration:(pressDuration/2)
-                       animations:^{
-        self.transform = CGAffineTransformMakeScale(magnification, magnification);
-      }
-                       completion:^(__unused BOOL finished2) {
-      }];
-    }];
+                     completion:^(__unused BOOL finished1){}];
+  }
+}
+
+- (void)setUpTVGestureRecognizers {
+  UILongPressGestureRecognizer *pressRecognizer = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(handlePress:)];
+  pressRecognizer.allowedPressTypes = @[ @(UIPressTypeSelect) ];
+  pressRecognizer.minimumPressDuration = 0.0;
+  pressRecognizer.delegate = self; // Press recognizer allows other recognizers to run
+
+  [self addGestureRecognizer:pressRecognizer];
+  _pressRecognizer = pressRecognizer;
 
+  UILongPressGestureRecognizer *longPressRecognizer = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(handleLongPress:)];
+  longPressRecognizer.allowedPressTypes = @[ @(UIPressTypeSelect) ];
+  longPressRecognizer.minimumPressDuration = 0.5;
+
+  [self addGestureRecognizer:longPressRecognizer];
+  _longPressRecognizer = longPressRecognizer;
+}
+
+- (void)tearDownTVGestureRecognizers {
+  if (_pressRecognizer) {
+    [self removeGestureRecognizer:_pressRecognizer];
+    _pressRecognizer = nil;
+  }
+  if (_longPressRecognizer) {
+    [self removeGestureRecognizer:_longPressRecognizer];
+    _longPressRecognizer = nil;
   }
 }
 
+
 - (void)handlePress:(UIGestureRecognizer *)r
 {
   switch (r.state) {
-      case UIGestureRecognizerStateBegan:
+    case UIGestureRecognizerStateBegan:
       _eventEmitter->onPressIn();
-          break;
-      case UIGestureRecognizerStateEnded:
-      case UIGestureRecognizerStateCancelled:
-      [self animatePress];
+      [self animatePressIn];
+      break;
+    case UIGestureRecognizerStateCancelled:
+    case UIGestureRecognizerStateEnded:
+      [self animatePressOut];
       _eventEmitter->onPressOut();
-          break;
-      default:
-          break;
+      break;
+    default:
+      break;
+  }
+}
+
+- (void)handleLongPress:(UIGestureRecognizer *)r
+{
+  switch (r.state) {
+    case UIGestureRecognizerStateBegan:
+      [self sendLongSelectStartNotification];
+      break;
+    case UIGestureRecognizerStateEnded:
+    case UIGestureRecognizerStateCancelled:
+      [self sendLongSelectEndNotification];
+      break;
+    default:
+      break;
   }
 }
 
@@ -1040,16 +1093,9 @@ const CGFloat BACKGROUND_COLOR_ZPOSITION = -1024.0f;
   // `isTVSelectable`
   if (oldViewProps.isTVSelectable != newViewProps.isTVSelectable) {
     if (newViewProps.isTVSelectable && ![self isTVFocusGuide]) {
-      UILongPressGestureRecognizer *pressRecognizer = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(handlePress:)];
-      pressRecognizer.allowedPressTypes = @[ @(UIPressTypeSelect) ];
-      pressRecognizer.minimumPressDuration = 0;
-      
-      [self addGestureRecognizer:pressRecognizer];
-      _pressRecognizer = pressRecognizer;
+      [self setUpTVGestureRecognizers];
     } else {
-      if (_pressRecognizer) {
-        [self removeGestureRecognizer:_pressRecognizer];
-      }
+      [self tearDownTVGestureRecognizers];
     }
   }
   // `tvParallaxProperties
diff --git a/node_modules/react-native/React/Views/RCTTVView.h b/node_modules/react-native/React/Views/RCTTVView.h
index 8ac1931..7c6e3e7 100644
--- a/node_modules/react-native/React/Views/RCTTVView.h
+++ b/node_modules/react-native/React/Views/RCTTVView.h
@@ -12,7 +12,7 @@
 #import <React/RCTBridge.h>
 
 //  A RCTView with additional properties and methods for user interaction using the Apple TV focus engine.
-@interface RCTTVView : RCTView
+@interface RCTTVView : RCTView <UIGestureRecognizerDelegate>
 
 /**
  * TV event handlers
@@ -77,11 +77,6 @@
  */
 - (void)sendBlurNotification:(UIFocusUpdateContext *)context;
 
-/**
- * Send Select Notification to listeners
- */
-- (void)sendSelectNotification:(UIGestureRecognizer *)recognizer;
-
 /**
  * Adds Parallax Motion Effects if tvParallaxProperty is enabled
  */
diff --git a/node_modules/react-native/React/Views/RCTTVView.m b/node_modules/react-native/React/Views/RCTTVView.m
index 4a71f36..5d1d18e 100644
--- a/node_modules/react-native/React/Views/RCTTVView.m
+++ b/node_modules/react-native/React/Views/RCTTVView.m
@@ -23,6 +23,7 @@
 @implementation RCTTVView {
   __weak RCTBridge *_bridge;
   UILongPressGestureRecognizer * _pressRecognizer;
+  UILongPressGestureRecognizer * _longPressRecognizer;
   BOOL motionEffectsAdded;
   NSArray* focusDestinations;
   id<UIFocusItem> previouslyFocusedItem;
@@ -76,46 +77,70 @@ RCT_NOT_IMPLEMENTED(-(instancetype)initWithCoder : unused)
 {
   self->_isTVSelectable = isTVSelectable;
   if (isTVSelectable && ![self isTVFocusGuide]) {
-    UILongPressGestureRecognizer *pressRecognizer = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(handlePress:)];
-    pressRecognizer.allowedPressTypes = @[ @(UIPressTypeSelect) ];
-    pressRecognizer.minimumPressDuration = 0;
+    [self setUpTVGestureRecognizers];
+  } else {
+    [self tearDownTVGestureRecognizers];
+  }
+}
+
+// Press recognizer should allow long press recognizer to work (but not the reverse)
+- (BOOL)gestureRecognizer:(UIGestureRecognizer *)gestureRecognizer shouldRecognizeSimultaneouslyWithGestureRecognizer:(UIGestureRecognizer *)otherGestureRecognizer {
+  return gestureRecognizer == _pressRecognizer;
+}
 
-    _pressRecognizer = pressRecognizer;
+- (void)setUpTVGestureRecognizers {
+  UILongPressGestureRecognizer *pressRecognizer = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(handlePress:)];
+  pressRecognizer.allowedPressTypes = @[ @(UIPressTypeSelect) ];
+  pressRecognizer.minimumPressDuration = 0.0;
+  pressRecognizer.delegate = self; // Press recognizer allows other recognizers to run
 
-    [self addGestureRecognizer:_pressRecognizer];
-  } else {
-    if (_pressRecognizer) {
-      [self removeGestureRecognizer:_pressRecognizer];
-    }
+  [self addGestureRecognizer:pressRecognizer];
+  _pressRecognizer = pressRecognizer;
+
+  UILongPressGestureRecognizer *longPressRecognizer = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(handleLongPress:)];
+  longPressRecognizer.allowedPressTypes = @[ @(UIPressTypeSelect) ];
+  longPressRecognizer.minimumPressDuration = 0.5;
+
+  [self addGestureRecognizer:longPressRecognizer];
+  _longPressRecognizer = longPressRecognizer;
+}
+
+- (void)tearDownTVGestureRecognizers {
+  if (_pressRecognizer) {
+    [self removeGestureRecognizer:_pressRecognizer];
+    _pressRecognizer = nil;
+  }
+  if (_longPressRecognizer) {
+    [self removeGestureRecognizer:_longPressRecognizer];
+    _longPressRecognizer = nil;
   }
 }
 
-- (void)animatePress
+
+- (void)animatePressIn
 {
   if ([self.tvParallaxProperties[@"enabled"] boolValue] == YES) {
-    float magnification = [self.tvParallaxProperties[@"magnification"] floatValue];
     float pressMagnification = [self.tvParallaxProperties[@"pressMagnification"] floatValue];
-    
-    // Duration of press animation
     float pressDuration = [self.tvParallaxProperties[@"pressDuration"] floatValue];
+    [UIView animateWithDuration:(pressDuration/2)
+                     animations:^{
+      self.transform = CGAffineTransformMakeScale(pressMagnification, pressMagnification);
+    }
+                     completion:^(__unused BOOL finished1){}];
+  }
+}
     
-    // Delay of press animation
-    float pressDelay = [self.tvParallaxProperties[@"pressDelay"] floatValue];
-    
-    [[NSRunLoop currentRunLoop] runUntilDate:[NSDate dateWithTimeIntervalSinceNow:pressDelay]];
+- (void) animatePressOut
+{
+  if ([self.tvParallaxProperties[@"enabled"] boolValue] == YES) {
+    float magnification = [self.tvParallaxProperties[@"magnification"] floatValue];
+    float pressDuration = [self.tvParallaxProperties[@"pressDuration"] floatValue];
     
     [UIView animateWithDuration:(pressDuration/2)
                      animations:^{
-      self.transform = CGAffineTransformMakeScale(pressMagnification, pressMagnification);
+      self.transform = CGAffineTransformMakeScale(magnification, magnification);
     }
-                     completion:^(__unused BOOL finished1){
-      [UIView animateWithDuration:(pressDuration/2)
-                       animations:^{
-        self.transform = CGAffineTransformMakeScale(magnification, magnification);
-      }
-                       completion:^(__unused BOOL finished2) {
-      }];
-    }];
+                     completion:^(__unused BOOL finished1){}];
   }
 }
 
@@ -124,10 +149,11 @@ RCT_NOT_IMPLEMENTED(-(instancetype)initWithCoder : unused)
   switch (r.state) {
       case UIGestureRecognizerStateBegan:
       if (self.onPressIn) self.onPressIn(nil);
+      [self animatePressIn];
           break;
       case UIGestureRecognizerStateEnded:
       case UIGestureRecognizerStateCancelled:
-      [self animatePress];
+      [self animatePressOut];
       if (self.onPressOut) self.onPressOut(nil);
           break;
       default:
@@ -135,6 +161,21 @@ RCT_NOT_IMPLEMENTED(-(instancetype)initWithCoder : unused)
   }
 }
 
+- (void)handleLongPress:(UIGestureRecognizer *)r
+{
+  switch (r.state) {
+    case UIGestureRecognizerStateBegan:
+      [self sendLongSelectStartNotification];
+      break;
+    case UIGestureRecognizerStateEnded:
+    case UIGestureRecognizerStateCancelled:
+      [self sendLongSelectEndNotification];
+      break;
+    default:
+      break;
+  }
+}
+
 - (BOOL)isTVFocusGuide
 {
   return self.focusGuide != nil;
@@ -441,14 +482,14 @@ RCT_NOT_IMPLEMENTED(-(instancetype)initWithCoder : unused)
     [[NSNotificationCenter defaultCenter] postNavigationBlurEventWithTag:self.reactTag target:self.reactTag];
 }
 
-- (void)sendSelectNotification:(UIGestureRecognizer *)recognizer
+- (void)sendLongSelectStartNotification
 {
-    [[NSNotificationCenter defaultCenter] postNavigationPressEventWithType:RCTTVRemoteEventSelect keyAction:RCTTVRemoteEventKeyActionUp tag:self.reactTag target:self.reactTag];
+    [[NSNotificationCenter defaultCenter] postNavigationPressEventWithType:RCTTVRemoteEventLongSelect keyAction:RCTTVRemoteEventKeyActionDown tag:self.reactTag target:self.reactTag];
 }
 
-- (void)sendLongSelectNotification:(UIGestureRecognizer *)recognizer
+- (void)sendLongSelectEndNotification
 {
-    [[NSNotificationCenter defaultCenter] postNavigationPressEventWithType:RCTTVRemoteEventLongSelect keyAction:recognizer.eventKeyAction tag:self.reactTag target:self.reactTag];
+    [[NSNotificationCenter defaultCenter] postNavigationPressEventWithType:RCTTVRemoteEventLongSelect keyAction:RCTTVRemoteEventKeyActionUp tag:self.reactTag target:self.reactTag];
 }
 
 - (RCTTVView *)getViewById:(NSNumber *)viewId {
diff --git a/node_modules/react-native/types/public/ReactNativeTVTypes.d.ts b/node_modules/react-native/types/public/ReactNativeTVTypes.d.ts
index 8cc7a91..d18dcce 100644
--- a/node_modules/react-native/types/public/ReactNativeTVTypes.d.ts
+++ b/node_modules/react-native/types/public/ReactNativeTVTypes.d.ts
@@ -100,7 +100,7 @@ declare module 'react-native' {
     pressDuration?: number | undefined,
 
     /**
-     * Defaults to 0.3
+     * @deprecated No longer used
      */
     pressDelay?: number | undefined,
   };
