diff --git a/node_modules/react-native/sdks/hermes-engine/hermes-engine.podspec b/node_modules/react-native/sdks/hermes-engine/hermes-engine.podspec
index d2d4141..49da687 100644
--- a/node_modules/react-native/sdks/hermes-engine/hermes-engine.podspec
+++ b/node_modules/react-native/sdks/hermes-engine/hermes-engine.podspec
@@ -8,7 +8,7 @@ require_relative "./hermes-utils.rb"
 
 react_native_path = File.dirname(Pod::Executable.execute_command('node', ['-p',
   'require.resolve(
-  "react-native-tvos",
+  "react-native",
     {paths: [process.argv[1]]},
   )', __dir__]).strip
 )
