diff --git a/node_modules/react-native-gesture-handler/apple/Handlers/RNPanHandler.m b/node_modules/react-native-gesture-handler/apple/Handlers/RNPanHandler.m
index 550777c..8c3f9a3 100644
--- a/node_modules/react-native-gesture-handler/apple/Handlers/RNPanHandler.m
+++ b/node_modules/react-native-gesture-handler/apple/Handlers/RNPanHandler.m
@@ -455,8 +455,6 @@
 #else
 - (RNGestureHandlerEventExtraData *)eventExtraData:(UIPanGestureRecognizer *)recognizer
 {
-  RNBetterPanGestureRecognizer *panRecognizer = (RNBetterPanGestureRecognizer *)recognizer;
-
   return [RNGestureHandlerEventExtraData
                     forPan:[recognizer locationInView:recognizer.view]
       withAbsolutePosition:[recognizer locationInView:recognizer.view.window]
@@ -464,7 +462,7 @@
               withVelocity:[recognizer velocityInView:recognizer.view.window]
        withNumberOfTouches:recognizer.numberOfTouches
            withPointerType:_pointerType
-            withStylusData:[panRecognizer.stylusData toDictionary]]; // In Objective-C calling method on nil returns
+            withStylusData:nil]; // In Objective-C calling method on nil returns
                                                                      // nil, therefore this line does not crash.
 }
 #endif
